lib/internal/fs/promises.js

```diff
@@ -296,9 +296,16 @@ async function writeFileHandle(filehandle, data, signal, encoding) {
   if (isCustomIterable(data)) {
     for await (const buf of data) {
       checkAborted(signal);
-      await write(
-        filehandle, buf, undefined,
-        isArrayBufferView(buf) ? buf.byteLength : encoding);
+      const toWrite =
+        isArrayBufferView(buf) ? buf : Buffer.from(buf, encoding || 'utf8');
+      let remaining = toWrite.byteLength;
+      while (remaining > 0) {
+        checkAborted(signal);
+        const writeSize = MathMin(kWriteFileMaxChunkSize, remaining);
+        const { bytesWritten } = await write(
+          filehandle, toWrite, toWrite.byteLength - remaining, writeSize);
+        remaining -= bytesWritten;
+      }
       checkAborted(signal);
```

Comment:
The execution is synchronous from the beginning of the `for` loop block to the beginning of the `while` loop block. I think it's safe to remove one check.
```suggestion
        const writeSize = MathMin(kWriteFileMaxChunkSize, remaining);
        const { bytesWritten } = await write(
          filehandle, toWrite, toWrite.byteLength - remaining, writeSize);
        remaining -= bytesWritten;
        checkAborted(signal);
      }
```

Reply:
It's safe, but I wanted to avoid the buffer allocation if the signal was already aborted, so I kept it there.

Reply:
>  I wanted to avoid the possible buffer allocation if the signal was already aborted, so I kept it there.

Not sure what you mean by that... Either the signal is aborted at line 298, and `checkAborted ` throws (so no buffer is allocated), either the signal is not aborted and the buffer needs to be allocated. Is there a third option?